packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -737,9 +741,52 @@ class InspectorController extends DisposableController
       parent.endShowNode();
     }
 
+    _updateSelectedErrorFromNode(_selectedNode.value);
     animateTo(selectedNode.value);
   }
 
+  /// Update the index of the selected error based on a node that has been
+  /// selected in the tree.
+  void _updateSelectedErrorFromNode(InspectorTreeNode node) {
+    final inspectorRef = node?.diagnostic?.valueRef?.id;
+
+    final errors = serviceManager.errorBadgeManager
+        .erroredItemsForPage(InspectorScreen.id)
+        .value;
+
+    // Check whether the node that was just selected has any errors associated
+    // with it.
+    var errorIndex = inspectorRef != null
+        ? errors.keys.toList().indexOf(inspectorRef)
+        : null;
+    if (errorIndex == -1) {
+      errorIndex = null;
+    }
+
+    _selectedErrorIndex.value = errorIndex;
+
+    // Additionally, mark this error as "read" and clear the badge.
+    if (errorIndex != null) {
+      serviceManager.errorBadgeManager.clearErrors(InspectorScreen.id);
```

Comment:
Can you add a comment explaining why this clearErrors is needed?

Reply:
Done! (46d17542)